### PR TITLE
Added toString method in weakKey class.

### PR DESCRIPTION
**Pull request by FuncGuy**
Added toString method in weakKey class so that whenever the map is printed key's actual value will be printed rather than object internal address.

### CODE REVIEW

File: src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentMap.java [hunk 1]
```diff
@@ -298,8 +300,11 @@ public void run() {
 
         private final int hashCode;
 
-        WeakKey(T key, ReferenceQueue<? super T> queue) {
+        private T key;
```
Comment (felixbarny) [inline src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentMap.java]: This creates a strong reference to the key, thus it's not actually a `WeakReference` anymore.

File: src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentMap.java [hunk 2]
```diff
@@ -316,6 +321,11 @@ public boolean equals(Object other) {
                 return ((WeakKey<?>) other).get() == get();
             }
         }
+
+        @Override
+        public String toString(){
+            return valueOf(key);
```
Comment (felixbarny) [inline src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentMap.java]:
```suggestion
            return valueOf(get());
```

**Review by felixbarny — CHANGES_REQUESTED** [review]

**Comment by felixbarny** [comment]
Also, I think you'll need to override the `toString` method with:

```java
public String toString() {
    return target.toString();
}
```

File: src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentMap.java (hunk 1 above)
Comment (FuncGuy) [reply src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentMap.java]: Thanks for pointing it !!

**Comment by FuncGuy** [comment]
@felixbarny calling target.toString() will not work because WeakKey is a static class but target is non static field.

**Comment by felixbarny** [comment]
What I meant is to add that method in the parent class, `WeakConcurrentMap`.

**Comment by FuncGuy** [comment]
@felixbarny if i move toString() method to WeakConcurrentMap then i won't be able to access get() method which is present in the WeakReference class and WeakConcurrentMap doesn't extends WeakReference.

**Comment by felixbarny** [comment]
I'm not suggesting to move the `toString` method from `WeakConcurrentMap$WeakKey` to `WeakConcurrentMap`. Instead, I suggest to add this method to `WeakConcurrentMap` in addition to the one you have added:

```java
public String toString() {
    return target.toString();
}
```

Otherwise, `new WeakConcurrentMap().toString()` would still just return the class name and the identity hash code.

**Comment by FuncGuy** [comment]
ha! got it !!

**Review by felixbarny — APPROVED** [review]
LGTM

**Pull request merged**